packages/adapter-utils/src/compare.ts

```diff
@@ -377,6 +378,14 @@ const filterChangesForApply = (changes: DetailedChange[]): DetailedChange[] => {
  * function in a single call
  */
 export const applyDetailedChanges = (element: ChangeDataType, detailedChanges: DetailedChange[]): void => {
+  if (detailedChanges.length === 1) {
+    const change = detailedChanges[0]
+    if (isModificationChange(change) && change.id.isTopLevel()) {
```

Comment:
I think this is the wrong check, since `ChangeDataType` could be a field (and thus, not a top level ID)
it should be
```suggestion
    if (change.id.isEqual(element.elemID)) {
```

and then I wonder about the check for `isModificationChange`
we don't really support the case where we get a removal or addition on the ID of the element, should we fail more explicitly in that case?

Reply:
The field case is covered by UT and works, is there a case you think is missing?
Anyway, went with your suggestion abut kept the `isModificationChange` test just to stay on the safe side and make it clear that we don't cover additions and removals in this flow.

Reply:
couldn't find the UT where a field is passed as the `element`, looks like in all the new UTs the `element` is an `ObjectType` (and I assume old UTs didn't cover it since it would not have worked before this PR).

Reply:
Is that even allowed? I thought there's no flow that does this.

Reply:
Done.